<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>7 QC HTML report | SingleCellMQC</title>
  <meta name="description" content="<p>This is a minimal example of using the bookdown package to write a book.
The HTML output format for this example is bookdown::bs4_book,
set in the _output.yml file.</p>" />
  <meta name="generator" content="bookdown 0.41 and GitBook 2.6.7" />

  <meta property="og:title" content="7 QC HTML report | SingleCellMQC" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="<p>This is a minimal example of using the bookdown package to write a book.
The HTML output format for this example is bookdown::bs4_book,
set in the _output.yml file.</p>" />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="7 QC HTML report | SingleCellMQC" />
  
  <meta name="twitter:description" content="<p>This is a minimal example of using the bookdown package to write a book.
The HTML output format for this example is bookdown::bs4_book,
set in the _output.yml file.</p>" />
  

<meta name="author" content="Daihan Ji#, Mei Han#, Shuting Lu, Jiaying Zeng, Wen Zhong*" />


<meta name="date" content="2025-05-08" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="batch-effect-evaluation.html"/>
<link rel="next" href="tips.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<script src="libs/core-js-2.5.3/shim.min.js"></script>
<script src="libs/react-18.2.0/react.min.js"></script>
<script src="libs/react-18.2.0/react-dom.min.js"></script>
<script src="libs/reactwidget-1.0.0/react-tools.js"></script>
<script src="libs/htmlwidgets-1.6.4/htmlwidgets.js"></script>
<link href="libs/reactable-0.4.4/reactable.css" rel="stylesheet" />
<script src="libs/reactable-binding-0.4.4/reactable.js"></script>
<link href="libs/jquery-sparkline-2.1.2/jquery.sparkline.css" rel="stylesheet" />
<script src="libs/jquery-sparkline-2.1.2/jquery.sparkline.js"></script>
<script src="libs/sparkline-binding-2.0/sparkline.js"></script>
<script src="libs/plotly-binding-4.10.4/plotly.js"></script>
<script src="libs/typedarray-0.1/typedarray.min.js"></script>
<link href="libs/crosstalk-1.2.1/css/crosstalk.min.css" rel="stylesheet" />
<script src="libs/crosstalk-1.2.1/js/crosstalk.min.js"></script>
<link href="libs/plotly-htmlwidgets-css-2.11.1/plotly-htmlwidgets.css" rel="stylesheet" />
<script src="libs/plotly-main-2.11.1/plotly-latest.min.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> <strong>About</strong></a>
<ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#introduction"><i class="fa fa-check"></i><b>1.1</b> Introduction</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#installation"><i class="fa fa-check"></i><b>1.2</b> Installation</a></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#bugs"><i class="fa fa-check"></i><b>1.3</b> Bugs</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="load-and-add-data.html"><a href="load-and-add-data.html"><i class="fa fa-check"></i><b>2</b> <strong>Load and add data</strong></a>
<ul>
<li class="chapter" data-level="2.1" data-path="load-and-add-data.html"><a href="load-and-add-data.html#loading-data-of-scrna-seq-or-cite-seq"><i class="fa fa-check"></i><b>2.1</b> Loading data of scRNA-seq or CITE-seq</a>
<ul>
<li class="chapter" data-level="2.1.1" data-path="load-and-add-data.html"><a href="load-and-add-data.html#data-import-without-bpcells"><i class="fa fa-check"></i><b>2.1.1</b> Data import Without BPCells</a></li>
<li class="chapter" data-level="2.1.2" data-path="load-and-add-data.html"><a href="load-and-add-data.html#data-import-with-bpcells"><i class="fa fa-check"></i><b>2.1.2</b> Data import With BPCells</a></li>
</ul></li>
<li class="chapter" data-level="2.2" data-path="load-and-add-data.html"><a href="load-and-add-data.html#loading-data-of-scvdj-seq"><i class="fa fa-check"></i><b>2.2</b> Loading data of scV(D)J-seq</a></li>
<li class="chapter" data-level="2.3" data-path="load-and-add-data.html"><a href="load-and-add-data.html#loading-data-of-multi-omics-scrna-seqcite-seq-scvdj-seq"><i class="fa fa-check"></i><b>2.3</b> Loading data of multi-omics (scRNA-seq/CITE-seq + scV(D)J-seq)</a></li>
<li class="chapter" data-level="2.4" data-path="load-and-add-data.html"><a href="load-and-add-data.html#loading-and-adding-10x-metrics"><i class="fa fa-check"></i><b>2.4</b> Loading and adding 10X metrics</a></li>
<li class="chapter" data-level="2.5" data-path="load-and-add-data.html"><a href="load-and-add-data.html#adding-information-of-samples"><i class="fa fa-check"></i><b>2.5</b> Adding information of samples</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="sample-qc.html"><a href="sample-qc.html"><i class="fa fa-check"></i><b>3</b> <strong>Sample QC</strong></a>
<ul>
<li class="chapter" data-level="3.1" data-path="sample-qc.html"><a href="sample-qc.html#sample-quality-assessment"><i class="fa fa-check"></i><b>3.1</b> Sample quality assessment</a>
<ul>
<li class="chapter" data-level="3.1.1" data-path="sample-qc.html"><a href="sample-qc.html#x-metrics-assessment"><i class="fa fa-check"></i><b>3.1.1</b> 10X metrics assessment</a></li>
<li class="chapter" data-level="3.1.2" data-path="sample-qc.html"><a href="sample-qc.html#basic-metrics-assessment"><i class="fa fa-check"></i><b>3.1.2</b> Basic metrics assessment</a></li>
<li class="chapter" data-level="3.1.3" data-path="sample-qc.html"><a href="sample-qc.html#visualisation-of-vdj-data"><i class="fa fa-check"></i><b>3.1.3</b> Visualisation of V(D)J data</a></li>
</ul></li>
<li class="chapter" data-level="3.2" data-path="sample-qc.html"><a href="sample-qc.html#pct-outlier-sample-detection"><i class="fa fa-check"></i><b>3.2</b> PCT outlier sample detection</a>
<ul>
<li class="chapter" data-level="3.2.1" data-path="sample-qc.html"><a href="sample-qc.html#cell-type-annotation-sctype"><i class="fa fa-check"></i><b>3.2.1</b> Cell type annotation (ScType)</a></li>
<li class="chapter" data-level="3.2.2" data-path="sample-qc.html"><a href="sample-qc.html#comparison-of-predicted-results-with-cell-proportion-range-database"><i class="fa fa-check"></i><b>3.2.2</b> Comparison of predicted results with cell proportion range database</a></li>
<li class="chapter" data-level="3.2.3" data-path="sample-qc.html"><a href="sample-qc.html#inter-sample-outliers"><i class="fa fa-check"></i><b>3.2.3</b> Inter-sample outliers</a></li>
<li class="chapter" data-level="3.2.4" data-path="sample-qc.html"><a href="sample-qc.html#outlier-detection-for-metrics-with-different-cell-types"><i class="fa fa-check"></i><b>3.2.4</b> Outlier detection for metrics with different cell types</a></li>
<li class="chapter" data-level="3.2.5" data-path="sample-qc.html"><a href="sample-qc.html#visualisation-of-the-pct-of-cell-types"><i class="fa fa-check"></i><b>3.2.5</b> Visualisation of the PCT of cell types</a></li>
</ul></li>
<li class="chapter" data-level="3.3" data-path="sample-qc.html"><a href="sample-qc.html#sample-identity-verification"><i class="fa fa-check"></i><b>3.3</b> Sample identity verification</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="cell-qc.html"><a href="cell-qc.html"><i class="fa fa-check"></i><b>4</b> <strong>Cell QC</strong></a>
<ul>
<li class="chapter" data-level="4.1" data-path="cell-qc.html"><a href="cell-qc.html#empty-droplets-detection-if-required"><i class="fa fa-check"></i><b>4.1</b> Empty droplets detection (If Required)</a></li>
<li class="chapter" data-level="4.2" data-path="cell-qc.html"><a href="cell-qc.html#low-quality-cells-detection"><i class="fa fa-check"></i><b>4.2</b> Low quality cells detection</a>
<ul>
<li class="chapter" data-level="4.2.1" data-path="cell-qc.html"><a href="cell-qc.html#fixed-cutoff"><i class="fa fa-check"></i><b>4.2.1</b> Fixed cutoff</a></li>
<li class="chapter" data-level="4.2.2" data-path="cell-qc.html"><a href="cell-qc.html#mad-method"><i class="fa fa-check"></i><b>4.2.2</b> MAD method</a></li>
<li class="chapter" data-level="4.2.3" data-path="cell-qc.html"><a href="cell-qc.html#miqc-method"><i class="fa fa-check"></i><b>4.2.3</b> miQC method</a></li>
<li class="chapter" data-level="4.2.4" data-path="cell-qc.html"><a href="cell-qc.html#ddqc-method"><i class="fa fa-check"></i><b>4.2.4</b> ddqc method</a></li>
<li class="chapter" data-level="4.2.5" data-path="cell-qc.html"><a href="cell-qc.html#run-multiple-methods"><i class="fa fa-check"></i><b>4.2.5</b> Run multiple methods</a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="cell-qc.html"><a href="cell-qc.html#check-low-quality-cells-optional"><i class="fa fa-check"></i><b>4.3</b> Check low-quality cells (optional)</a></li>
<li class="chapter" data-level="4.4" data-path="cell-qc.html"><a href="cell-qc.html#doublets-detection"><i class="fa fa-check"></i><b>4.4</b> Doublets detection</a>
<ul>
<li class="chapter" data-level="4.4.1" data-path="cell-qc.html"><a href="cell-qc.html#scdblfinder-method"><i class="fa fa-check"></i><b>4.4.1</b> scDblFinder method</a></li>
<li class="chapter" data-level="4.4.2" data-path="cell-qc.html"><a href="cell-qc.html#hybrid-method"><i class="fa fa-check"></i><b>4.4.2</b> hybrid method</a></li>
<li class="chapter" data-level="4.4.3" data-path="cell-qc.html"><a href="cell-qc.html#bcds-method"><i class="fa fa-check"></i><b>4.4.3</b> bcds method</a></li>
<li class="chapter" data-level="4.4.4" data-path="cell-qc.html"><a href="cell-qc.html#cxds-method"><i class="fa fa-check"></i><b>4.4.4</b> cxds method</a></li>
<li class="chapter" data-level="4.4.5" data-path="cell-qc.html"><a href="cell-qc.html#doubletfinder-method"><i class="fa fa-check"></i><b>4.4.5</b> DoubletFinder method</a></li>
<li class="chapter" data-level="4.4.6" data-path="cell-qc.html"><a href="cell-qc.html#vdj-method"><i class="fa fa-check"></i><b>4.4.6</b> VDJ method</a></li>
<li class="chapter" data-level="4.4.7" data-path="cell-qc.html"><a href="cell-qc.html#adt-method"><i class="fa fa-check"></i><b>4.4.7</b> ADT method</a></li>
<li class="chapter" data-level="4.4.8" data-path="cell-qc.html"><a href="cell-qc.html#run-multiple-methods-1"><i class="fa fa-check"></i><b>4.4.8</b> Run multiple methods</a></li>
</ul></li>
<li class="chapter" data-level="4.5" data-path="cell-qc.html"><a href="cell-qc.html#visualization-of-cell-qc-results"><i class="fa fa-check"></i><b>4.5</b> Visualization of cell QC results</a>
<ul>
<li class="chapter" data-level="4.5.1" data-path="cell-qc.html"><a href="cell-qc.html#visualization-of-the-percentage-of-cells-filtered-by-different-methods"><i class="fa fa-check"></i><b>4.5.1</b> Visualization of the percentage of cells filtered by different methods</a></li>
<li class="chapter" data-level="4.5.2" data-path="cell-qc.html"><a href="cell-qc.html#overlap-between-different-methods-upset-plot"><i class="fa fa-check"></i><b>4.5.2</b> Overlap between different methods (UpSet plot)</a></li>
<li class="chapter" data-level="4.5.3" data-path="cell-qc.html"><a href="cell-qc.html#comparison-of-metrics-between-pass-and-fail-cells-identified-by-methods"><i class="fa fa-check"></i><b>4.5.3</b> Comparison of metrics between <em>Pass</em> and <em>Fail</em> cells identified by methods</a></li>
<li class="chapter" data-level="4.5.4" data-path="cell-qc.html"><a href="cell-qc.html#visualization-of-scatter-plots-for-cell-metrics"><i class="fa fa-check"></i><b>4.5.4</b> Visualization of scatter plots for cell metrics</a></li>
</ul></li>
<li class="chapter" data-level="4.6" data-path="cell-qc.html"><a href="cell-qc.html#filter-out-problematic-droplets-or-cells"><i class="fa fa-check"></i><b>4.6</b> Filter out problematic droplets or cells</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="feature-qc.html"><a href="feature-qc.html"><i class="fa fa-check"></i><b>5</b> <strong>Feature QC</strong></a>
<ul>
<li class="chapter" data-level="5.1" data-path="feature-qc.html"><a href="feature-qc.html#feature-quality-assessment"><i class="fa fa-check"></i><b>5.1</b> Feature quality assessment</a>
<ul>
<li class="chapter" data-level="5.1.1" data-path="feature-qc.html"><a href="feature-qc.html#calculating-feature-metrics"><i class="fa fa-check"></i><b>5.1.1</b> Calculating feature metrics</a></li>
<li class="chapter" data-level="5.1.2" data-path="feature-qc.html"><a href="feature-qc.html#reference-qc-thresholds-for-feature-qc"><i class="fa fa-check"></i><b>5.1.2</b> Reference QC thresholds for feature QC</a></li>
<li class="chapter" data-level="5.1.3" data-path="feature-qc.html"><a href="feature-qc.html#calculating-per-feature-variance-explained-by-batch-factors"><i class="fa fa-check"></i><b>5.1.3</b> Calculating per-feature variance explained by batch factors</a></li>
<li class="chapter" data-level="5.1.4" data-path="feature-qc.html"><a href="feature-qc.html#base-plots-of-feature-metrics"><i class="fa fa-check"></i><b>5.1.4</b> Base plots of feature metrics</a></li>
<li class="chapter" data-level="5.1.5" data-path="feature-qc.html"><a href="feature-qc.html#scatter-plot-of-the-feature-metrics"><i class="fa fa-check"></i><b>5.1.5</b> Scatter plot of the feature metrics</a></li>
</ul></li>
<li class="chapter" data-level="5.2" data-path="feature-qc.html"><a href="feature-qc.html#systematic-background-noise-correction"><i class="fa fa-check"></i><b>5.2</b> Systematic background noise correction</a>
<ul>
<li class="chapter" data-level="5.2.1" data-path="feature-qc.html"><a href="feature-qc.html#sccdc"><i class="fa fa-check"></i><b>5.2.1</b> scCDC</a></li>
<li class="chapter" data-level="5.2.2" data-path="feature-qc.html"><a href="feature-qc.html#decontx-rna"><i class="fa fa-check"></i><b>5.2.2</b> DecontX (RNA)</a></li>
<li class="chapter" data-level="5.2.3" data-path="feature-qc.html"><a href="feature-qc.html#decontpro-adt"><i class="fa fa-check"></i><b>5.2.3</b> DecontPro (ADT)</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="6" data-path="batch-effect-evaluation.html"><a href="batch-effect-evaluation.html"><i class="fa fa-check"></i><b>6</b> <strong>Batch effect evaluation</strong></a>
<ul>
<li class="chapter" data-level="6.1" data-path="batch-effect-evaluation.html"><a href="batch-effect-evaluation.html#sample-level"><i class="fa fa-check"></i><b>6.1</b> Sample level</a>
<ul>
<li class="chapter" data-level="6.1.1" data-path="batch-effect-evaluation.html"><a href="batch-effect-evaluation.html#pca-of-pseudobulk-data-at-the-sample-level"><i class="fa fa-check"></i><b>6.1.1</b> PCA of pseudobulk data at the sample level</a></li>
<li class="chapter" data-level="6.1.2" data-path="batch-effect-evaluation.html"><a href="batch-effect-evaluation.html#performing-svd-analysis-and-calculating-r²-values"><i class="fa fa-check"></i><b>6.1.2</b> Performing SVD analysis and calculating R² values</a></li>
</ul></li>
<li class="chapter" data-level="6.2" data-path="batch-effect-evaluation.html"><a href="batch-effect-evaluation.html#cell-level"><i class="fa fa-check"></i><b>6.2</b> Cell level</a>
<ul>
<li class="chapter" data-level="6.2.1" data-path="batch-effect-evaluation.html"><a href="batch-effect-evaluation.html#rna-umap"><i class="fa fa-check"></i><b>6.2.1</b> RNA UMAP</a></li>
<li class="chapter" data-level="6.2.2" data-path="batch-effect-evaluation.html"><a href="batch-effect-evaluation.html#adt-umap"><i class="fa fa-check"></i><b>6.2.2</b> ADT UMAP</a></li>
<li class="chapter" data-level="6.2.3" data-path="batch-effect-evaluation.html"><a href="batch-effect-evaluation.html#rnaadt-umap"><i class="fa fa-check"></i><b>6.2.3</b> RNA+ADT UMAP</a></li>
<li class="chapter" data-level="6.2.4" data-path="batch-effect-evaluation.html"><a href="batch-effect-evaluation.html#harmony-umap"><i class="fa fa-check"></i><b>6.2.4</b> harmony UMAP</a></li>
</ul></li>
<li class="chapter" data-level="6.3" data-path="batch-effect-evaluation.html"><a href="batch-effect-evaluation.html#feature-level"><i class="fa fa-check"></i><b>6.3</b> Feature level</a>
<ul>
<li class="chapter" data-level="6.3.1" data-path="batch-effect-evaluation.html"><a href="batch-effect-evaluation.html#calculating-per-feature-variance-explained-by-variables-at-cell-level"><i class="fa fa-check"></i><b>6.3.1</b> Calculating per-feature variance explained by variables at cell level</a></li>
<li class="chapter" data-level="6.3.2" data-path="batch-effect-evaluation.html"><a href="batch-effect-evaluation.html#calculating-per-feature-variance-explained-by-variables-at-sample-level"><i class="fa fa-check"></i><b>6.3.2</b> Calculating per-feature variance explained by variables at sample level</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="7" data-path="qc-html-report.html"><a href="qc-html-report.html"><i class="fa fa-check"></i><b>7</b> <strong>QC HTML report</strong></a></li>
<li class="chapter" data-level="8" data-path="tips.html"><a href="tips.html"><i class="fa fa-check"></i><b>8</b> <strong>Tips</strong></a>
<ul>
<li class="chapter" data-level="8.1" data-path="tips.html"><a href="tips.html#progress-updates"><i class="fa fa-check"></i><b>8.1</b> Progress updates</a></li>
<li class="chapter" data-level="8.2" data-path="tips.html"><a href="tips.html#future-parallel"><i class="fa fa-check"></i><b>8.2</b> Future parallel</a></li>
</ul></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">SingleCellMQC</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="qc-html-report" class="section level1 hasAnchor" number="7">
<h1><span class="header-section-number">7</span> <strong>QC HTML report</strong><a href="qc-html-report.html#qc-html-report" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p><strong>Compiled: 2025-05-08</strong>
<strong>Written by Daihan Ji</strong></p>
<blockquote>
<p><strong>If you have already performed preprocessing, you can directly run the <code>RunReport</code> function to generate a QC web report. If not, you can run <code>RunPreprocess</code> to complete some basic workflows.</strong></p>
</blockquote>
<p><strong>Function Descriptions: <code>RunPreprocess</code> and <code>RunReport</code>:</strong></p>
<p><strong>1. <code>RunPreprocess</code> Function</strong></p>
<p><code>RunPreprocess</code> is a function for data preprocessing and includes the following main features:</p>
<ul>
<li><p><strong>Calculate Quality Metrics</strong>: Adds quality metrics (e.g., number of genes, mitochondrial percentage) to the metadata.</p></li>
<li><p><strong>Cell Type Annotation</strong>: Runs cell type annotation using the <code>RunScType</code> function.</p></li>
<li><p><strong>Doublet Detection</strong>: Detects doublets using specified methods (e.g., <code>scDblFinder</code>).</p></li>
<li><p><strong>Low-Quality Cell Detection</strong>: Filters low-quality cells using specified methods (e.g., MAD).</p></li>
<li><p><strong>Dimensionality Reduction</strong>: Runs a preprocessing pipeline (e.g., UMAP) for visualization and downstream analysis.</p></li>
</ul>
<p><strong>2. <code>RunReport</code> Function</strong></p>
<p><code>RunReport</code> allows you to freely choose the sections of the report you want to generate. The available sections are:</p>
<p><strong>Sample QC Module</strong></p>
<ul>
<li><p><strong><code>section.sample</code></strong>: Sections to include in the Sample QC module. Default is <code>c(1:3)</code>.</p>
<ul>
<li><p><strong>1</strong>: Run Sample QC: Sample quality assessment.</p></li>
<li><p><strong>2</strong>: Run Sample QC: Outlier sample detection by cell type.</p></li>
<li><p><strong>3</strong>: Run Sample QC: Sample identity verification.</p></li>
</ul></li>
</ul>
<p><strong>Cell QC Module</strong></p>
<ul>
<li><p><strong><code>section.cell</code></strong>: Sections to include in the Cell QC module. Default is <code>c(1:2)</code>.</p>
<ul>
<li><p><strong>1</strong>: Run Cell QC: Low-quality cell information.</p></li>
<li><p><strong>2</strong>: Run Cell QC: Doublet information.</p></li>
</ul></li>
</ul>
<p><strong>Feature QC Module</strong></p>
<ul>
<li><p><strong><code>section.feature</code></strong>: Sections to include in the Feature QC module. Default is <code>1</code>.</p>
<ul>
<li><strong>1</strong>: Run Feature QC: Feature quality assessment.</li>
</ul></li>
</ul>
<p><strong>Batch QC Module</strong></p>
<ul>
<li><p><strong><code>section.batch</code></strong>: Sections to include in the Batch QC module. Default is <code>1:3</code>.</p>
<ul>
<li><p><strong>1</strong>: Run Batch QC: Sample level.</p></li>
<li><p><strong>2</strong>: Run Batch QC: Cell level.</p></li>
<li><p><strong>3</strong>: Run Batch QC: Feature level.</p></li>
</ul></li>
</ul>
<hr />
<ul>
<li><strong>Load data</strong></li>
</ul>
<div class="sourceCode" id="cb151"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb151-1"><a href="qc-html-report.html#cb151-1" tabindex="-1"></a><span class="fu">library</span>(SingleCellMQC)</span>
<span id="cb151-2"><a href="qc-html-report.html#cb151-2" tabindex="-1"></a><span class="fu">library</span>(Seurat)</span>
<span id="cb151-3"><a href="qc-html-report.html#cb151-3" tabindex="-1"></a><span class="fu">library</span>(BPCells)</span>
<span id="cb151-4"><a href="qc-html-report.html#cb151-4" tabindex="-1"></a><span class="co">#Load a pre-existing Seurat object or processed raw data using SingleCellMQC.</span></span>
<span id="cb151-5"><a href="qc-html-report.html#cb151-5" tabindex="-1"></a>pbmc <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">&quot;/data/pbmc.rds&quot;</span>)</span></code></pre></div>
<ul>
<li><strong>Run Preprocessing Pipeline for Single-Cell Data (If no prior preprocessing has been performed)</strong></li>
</ul>
<div class="sourceCode" id="cb152"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb152-1"><a href="qc-html-report.html#cb152-1" tabindex="-1"></a>pbmc <span class="ot">&lt;-</span> <span class="fu">RunPreprocess</span>(pbmc, <span class="at">sample.by =</span> <span class="st">&quot;orig.ident&quot;</span>)</span></code></pre></div>
<ul>
<li><strong>Generate QC HTML Report</strong></li>
</ul>
<div class="sourceCode" id="cb153"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb153-1"><a href="qc-html-report.html#cb153-1" tabindex="-1"></a><span class="fu">RunReport</span>(</span>
<span id="cb153-2"><a href="qc-html-report.html#cb153-2" tabindex="-1"></a>  <span class="at">object=</span>pbmc,</span>
<span id="cb153-3"><a href="qc-html-report.html#cb153-3" tabindex="-1"></a>  <span class="at">tissue =</span> <span class="st">&quot;blood&quot;</span>, <span class="co"># tissue for PCT outlier sample detection.</span></span>
<span id="cb153-4"><a href="qc-html-report.html#cb153-4" tabindex="-1"></a>  <span class="at">RNA.batch.by =</span> <span class="fu">c</span>(<span class="st">&quot;orig.ident&quot;</span>, <span class="st">&quot;Batch_EXP&quot;</span>, <span class="st">&quot;Time_point&quot;</span>), <span class="co">#Column name(s) in the metadata indicating RNA batch information (categorical variable) or other categorical variables for PCA or UMAP plot. Default is &quot;orig.ident&quot;. Only for: section.batch (1, 2).</span></span>
<span id="cb153-5"><a href="qc-html-report.html#cb153-5" tabindex="-1"></a>  <span class="at">RNA.batch.covariate =</span> <span class="fu">c</span>(<span class="st">&quot;nFeature_RNA&quot;</span>, <span class="st">&quot;percent.mt&quot;</span>, <span class="st">&quot;Batch_EXP&quot;</span>, <span class="st">&quot;Time_point&quot;</span>) <span class="co">#Covariates (can be categorical or continuous) for RNA SVD and ⁠Variance Explained⁠ analysis. Default is c(&quot;orig.ident&quot;, &quot;percent.mt&quot;, &quot;nFeature_RNA&quot;, &quot;nCount_RNA&quot;).Only for: section.batch (1, 3).</span></span>
<span id="cb153-6"><a href="qc-html-report.html#cb153-6" tabindex="-1"></a>)</span></code></pre></div>

</div>
            </section>

          </div>
        </div>
      </div>
<a href="batch-effect-evaluation.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="tips.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
